lib/viewmodels/new_comment_view_model.dart

```diff
@@ -82,7 +86,21 @@ class NewCommentViewModel
       voteScore: 0,
     );
 
-    await commentRepository.addComment(postId, commentData);
+    // Note: The addition of the comment unedr the post and the comment under the
+    // the user document are not done atomically.
+    // This is because integrating atomicity would require a complex refactor
+    // that does not add much value to the application.
+    // In the worst case, the link between the comment and the user is lost,
+    // but the database is still consistent and won't crash the application.
+    final commentId = await commentRepository.addComment(postId, commentData);
+
+    final userCommentData = UserCommentData(
+      commentId: commentId,
+      parentPostId: postId,
+      content: content,
+    );
+
+    await userCommentRepository.addUserComment(currentUserId, userCommentData);
```

Comment:
You add a reference here, but never delete it when the comment is deleted. In fact, I think that this is not the view-model's job to update this in the database, but directly the `CommentRepositoryService`'s. 
In other words, if I want to add another way to add comments to the code base, I don't want to have to duplicate this code; it should already be done by the model. If you follow this suggestion, make sure to adapt the documentation correctly: the `delete` function from the `UserCommentRepositoryService` should never be used, except by the delete `CommentRepositoryService` functions of. This is still a bit weird, maybe there is another way to encapsulate the delete function of `UserCommentRepositoryService` to make sure nobody can use it outside `CommentRepositoryService`. What do you think?

Reply:
After thinking about it, I think that you are totally right !
However, this will require a substancial refactor.
Thus, I will remove the viewmodel update from this PR and perform the refactor in another one.
Thanks for catching important architecture mistake :+1: 

I remove this from the viewmodel in f39f4a5adc13daaf26eb67fb741f58a9cd716815.

Reply:
Note: Those function have slightly modified after removing the `UserCommentIdFirestore`.

I actually need the second one (`createMockUserComment`) when I have to get a `UserCommentFirestore` but doesn't need to have it in the database.

This happens in `user_comment_firestore_test.dart` for instance.
Merging the two and keeping only `addComments`, I would need to create a repository and add the comments to the database to get the mock data. And I don't think it makes a lot of sense for this test.

Thus, I personnaly prefer to keep the two functions here if that's ok for you ?

Reply:
Nice, thank you for your work :)